src/record/record.js

```diff
@@ -345,10 +324,13 @@ Record.prototype._recoverRecord = function( remoteVersion, remoteData, message )
  * @private
  * @returns {void}
  */
-Record.prototype._onRecordRecovered = function( remoteVersion, error, data ) {
+Record.prototype._onRecordRecovered = function( remoteVersion, remoteData, error, data ) {
 	if( !error ) {
 		this.version = remoteVersion;
-		this.set( data );
+		if ( data !== this._$data && !utils.deepEquals( data, this._$data ) ) {
```

Comment:
@yasserf: I would argue that this should be:

```javascript
if ( data !== remoteData && !utils.deepEquals( data, remoteData ) ) {
```

However, the tests disagree with me. The current behaviour is strange to me, why would we send to the server when the resolved data is the same as the remoteData, but we don't send when the resolved data is the same as the local one?

Isn't this a bug?